text/0000-inferred-variant-constructor.md

```diff
@@ -0,0 +1,204 @@
+- Feature Name: `inferred_variant_constructor`
+- Start Date: 2017-03-11
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Allow the name (qualifier) of an enum variant to be elided in expressions and
+patterns whenever it can be inferred.
+
+# Motivation
+[motivation]: #motivation
+
+The main goal of this proposal is to improve ergonomics of enums in Rust by
+reducing verbosity.  Currently, there are two convenient ways to use enums.
+One would either:
+
+  - bring the enum type into scope (`use foolib::Foo;`), or
+  - bring the enum variants into scope (`use foolib::Foo::*;`).
+
+The former is less convenient than the latter, but the latter pollutes the
+namespace more.  One could always write enum variants fully qualified,
+`foolib::Foo::Alpha`, but that is even more tedious.
+
+This proposal intends to make variant expressions as convenient and
+pollution-free as field expressions today.  Whereas field expressions allow
+the user to extract fields of a struct without ever bringing the struct into
+scope, inferred variant constructors allow the user to construct/match
+variants of an enum without ever bringing the enum or its variants into scope.
+
+This could lower the usability barrier for enums, particularly for library
+users.  It would reduce the verbosity of enums to a level similar to
+dynamically typed languages, where strings (e.g. Python) or symbols
+(e.g. Lisps, Ruby) designate ad hoc variants
+
+The feature could encourage library writers to use domain-specific enums more
+often, rather than to re-purpose generic enums like `Option` or `Result`,
+leading to more readable code.  It would also allow library writers to use
```

Comment:
> The ship has long sailed with regard to Rust’s context-sensitivity because of the .field syntax. 

I'm a bit confused to what you refer to when you mean field syntax, as it doesnt make any sense. If you mean the ability to do `.sth` on expressions and then the sth field or sth function gets accessed, then it doesn't really serves your argument, in fact its the opposite. First, it doesn't really compare to enums, and if it did, it is an example *for* explicitness, not against it, as you always need something to apply it to. You can't say `.new()` out of the blue, you always need an expression for it. The equivalent to your proposal would be `_.new()` or something, but as I said before, it doesn't really compare.